File: solutions/csharp/chessgame/Move.cs
namespace ChessGame
{
    public class Move
    {
        public Piece Piece { get; }
        public int DestRow { get; }
        public int DestCol { get; }

        public Move(Piece piece, int destRow, int destCol)
        {
            Piece = piece;
            DestRow = destRow;
            DestCol = destCol;
        }
    }
}

File: solutions/golang/chessgame/move.go
package chessgame

type Move struct {
	piece   Piece
	destRow int
	destCol int
}

func NewMove(piece Piece, destRow, destCol int) *Move {
	return &Move{
		piece:   piece,
		destRow: destRow,
		destCol: destCol,
	}
}
